### PR TITLE
Do not allow staff to re-use passwords

### PR DESCRIPTION
**Pull request by doomspork**
As an added security measure now that we will begin integrating our own SSO for third parties is to enforce a password reuse policy for staff.

Additional staff features / requirements we should consider:

1. mandatory 2FA
1.  mandatory password changes every 90 days

### CODE REVIEW

File: lib/recognizer/accounts/user.ex [hunk 1]
```diff
@@ -114,9 +115,43 @@ defmodule Recognizer.Accounts.User do
...
+  end
+
+  defp validate_password_reuse(%{data: user} = changeset) do
+    current_password = get_field(changeset, :password)
+
+    previous_passwords =
+      user
+      |> Repo.preload(:previous_passwords)
+      |> Map.get(:previous_passwords)
+      |> Enum.take(6)
```
[the remark this answers is not in the document]
Comment (dbernheisel) [reply lib/recognizer/accounts/user.ex:142]: Is this supposed to be the 6 most recently-used passwords? I'm not seeing where order is guaranteed.

[the remark this answers is not in the document]
Comment (doomspork) [reply lib/recognizer/accounts/user.ex:142]: We use `put_assoc` below with this value and their latest, that should truncate any other records. I'll confirm with tests :+1:

**Comment by caemor** [comment]
>     2\. mandatory password changes every 90 days

I just stumbled upon this pull request and saw the mandatory password change every 90 days.
There are many studies that mandatory password changes reduce the quality of passwords and therefore this is not recommended anymore by most. Sources:
- NIST: https://stealthbits.com/blog/nist-password-guidelines/ (summary with a link to the full nist document)
- BSI: https://www.bsi.bund.de/SharedDocs/Downloads/DE/BSI/Grundschutz/Umsetzungshinweise/Umsetzungshinweise_2021/Umsetzungshinweis_zum_Baustein_ORP_4_Identitaets_und_Berechtigungsmanagement.html (in german)
- UK: https://assets.publishing.service.gov.uk/government/uploads/system/uploads/attachment_data/file/458857/Password_guidance_-_simplifying_your_approach.pdf
- Microsoft: https://arstechnica.com/information-technology/2019/06/microsoft-says-mandatory-password-changing-is-ancient-and-obsolete/
- ...

Generally it's recommend to only force new password if you detect conspicious behaviour or the password is known to be in a breach.

If you really want to increase the security level, enfore 2FA.

**Comment by doomspork** [comment]
Howdy @caemor 👋 

This requirement is for employees and isn’t something we’re willing to deviate from. While there are some articles against password changes like this, there’s plenty of documentation in favor of it.

Since these are employees with access to internal systems, we feel confident this requirement coupled with our password strength and required 2FA is the best approach. 

Thanks for your feedback!

**Pull request closed without merging**